sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -556,6 +603,56 @@ private static void verifyUnsplittableMethods(ErrorReporter errors, DoFnSignatur
     return DoFnSignature.SplitRestrictionMethod.create(m, restrictionT);
   }
 
+  private static ImmutableMap<String, TimerDeclaration> analyzeTimerDeclarations(
+      ErrorReporter errors, Class<?> fnClazz) {
+    Map<String, DoFnSignature.TimerDeclaration> declarations = new HashMap<>();
+    for (Field field : declaredFieldsWithAnnotation(DoFn.TimerId.class, fnClazz, DoFn.class)) {
+      String id = field.getAnnotation(DoFn.TimerId.class).value();
+      if (!validateTimerField(errors, declarations, id, field)) {
+        continue;
+      }
+      declarations.put(id, DoFnSignature.TimerDeclaration.create(id, field));
+    }
+
+    return ImmutableMap.copyOf(declarations);
+  }
+
+  /**
+   * Returns {@code true} if the field is valid, otherwise return {@code false} and adds
```

Comment:
Oh, I had assumed the point of the `ErrorReporter` was to gather errors. Without that, it seems like just a needless parameter. And upon inspection, why does it exist?